src/test/java/uk/gov/ons/census/casesvc/messaging/AddressReceiverIT.java

```diff
@@ -149,16 +149,114 @@ public void testInvalidAddress() throws Exception {
   }
 
   @Test
-  public void testAddressModifiedEventTypeLoggedOnly() throws Exception {
-    PayloadDTO payload = new PayloadDTO();
-    payload.setAddressModification(createTestAddressModifiedJson(TEST_CASE_ID));
+  public void testAddressModifiedEventType() throws Exception {
+    try (QueueSpy rhCaseQueueSpy = rabbitQueueHelper.listen(rhCaseQueue)) {
+      // GIVEN
+      EasyRandom easyRandom = new EasyRandom();
+      Case caze = easyRandom.nextObject(Case.class);
+      caze.setCaseId(TEST_CASE_ID);
+      caze.setSurvey("CENSUS");
+      caze.setUacQidLinks(null);
+      caze.setEvents(null);
+      caze.setAddressInvalid(false);
+      caze = caseRepository.saveAndFlush(caze);
 
-    testEventTypeLoggedOnly(
-        payload,
-        JsonHelper.convertObjectToJson(payload.getAddressModification()),
-        ADDRESS_MODIFIED,
-        EventType.ADDRESS_MODIFIED,
-        "Address modified");
+      ResponseManagementEvent managementEvent = new ResponseManagementEvent();
+      managementEvent.setEvent(new EventDTO());
+      managementEvent.getEvent().setDateTime(OffsetDateTime.now());
+      managementEvent.getEvent().setChannel("Test channel");
+      managementEvent.getEvent().setSource("Test source");
+      managementEvent.getEvent().setType(ADDRESS_MODIFIED);
+      managementEvent.setPayload(new PayloadDTO());
+      managementEvent.getPayload().setAddressModification(new AddressModification());
+      managementEvent
+          .getPayload()
+          .getAddressModification()
+          .setCollectionCase(new CollectionCaseCaseId());
+      managementEvent
+          .getPayload()
+          .getAddressModification()
+          .getCollectionCase()
+          .setId(caze.getCaseId());
+      managementEvent.getPayload().getAddressModification().setNewAddress(new Address());
+      managementEvent
+          .getPayload()
+          .getAddressModification()
+          .getNewAddress()
+          .setAddressLine1("modified address line 1");
+      managementEvent
+          .getPayload()
+          .getAddressModification()
+          .getNewAddress()
+          .setAddressLine2("modified address line 2");
+      managementEvent
+          .getPayload()
+          .getAddressModification()
+          .getNewAddress()
+          .setAddressLine3("modified address line 3");
+      managementEvent
+          .getPayload()
+          .getAddressModification()
+          .getNewAddress()
+          .setTownName("modified town name");
+      managementEvent
+          .getPayload()
+          .getAddressModification()
+          .getNewAddress()
+          .setOrganisationName("modified org name");
+      managementEvent
+          .getPayload()
+          .getAddressModification()
+          .getNewAddress()
+          .setEstabType("HOSPITAL");
```

Comment:
These billions of lines could be reduced to 6 by pulling out the `newAddress` object